src/main/java/io/odpf/firehose/sink/clickhouse/ClickhouseSinkFactory.java

```diff
@@ -0,0 +1,79 @@
+package io.odpf.firehose.sink.clickhouse;
+
+import com.clickhouse.client.ClickHouseCompression;
+import com.clickhouse.client.ClickHouseCredentials;
+import com.clickhouse.client.ClickHouseNode;
+import com.clickhouse.client.ClickHouseClient;
+import com.clickhouse.client.ClickHouseNodeSelector;
+import com.clickhouse.client.ClickHouseRequest;
+import com.clickhouse.client.ClickHouseProtocol;
+import com.clickhouse.client.config.ClickHouseClientOption;
+import io.odpf.firehose.config.ClickhouseSinkConfig;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.metrics.StatsDReporter;
+import io.odpf.firehose.proto.ProtoToFieldMapper;
+import io.odpf.firehose.sink.AbstractSink;
+import io.odpf.stencil.Parser;
+import io.odpf.stencil.client.StencilClient;
+import org.aeonbits.owner.ConfigFactory;
+
+import java.util.Map;
+
+/**
+ * Factory class to create the ClickhouseSink.
+ * <p>
+ * The consumer framework would reflectively instantiate this factory
+ * using the configurations supplied and invoke {@see #create(Map < String, String > configuration, StatsDReporter statsDReporter, StencilClient client)}
+ * to obtain the Clickhouse sink implementation.
+ */
+public class ClickhouseSinkFactory {
+    /**
+     * x
+     * Create Clickhouse sink.
+     *
+     * @param configuration  the configuration
+     * @param statsDReporter the statsd reporter
+     * @param stencilClient  the client
+     * @return the abstract sink
+     */
+    public static AbstractSink create(Map<String, String> configuration, StatsDReporter statsDReporter, StencilClient stencilClient) {
+        ClickhouseSinkConfig clickhouseSinkConfig = ConfigFactory.create(ClickhouseSinkConfig.class, configuration);
+
+        Instrumentation instrumentation = new Instrumentation(statsDReporter, ClickhouseSinkFactory.class);
+
+        ClickHouseNode server = ClickHouseNode.builder()
+                .host(clickhouseSinkConfig.getClickhouseHost())
+                .port(ClickHouseProtocol.HTTP, Integer.parseInt(clickhouseSinkConfig.getClickhousePort()))
+                .database(clickhouseSinkConfig.getClickhouseDatabase()).credentials(ClickHouseCredentials.fromUserAndPassword(
+                        clickhouseSinkConfig.getClickhouseUsername(), clickhouseSinkConfig.getClickhousePassword()))
+                .build();
+
+        ClickHouseCompression clickHouseCompression = null;
+
+        if (clickhouseSinkConfig.getClickhouseCompressionAlgorithm().equals("lz4")) {
+            clickHouseCompression = ClickHouseCompression.LZ4;
```

Comment:
instead of doing that, we can return ClickHouseCompression directly from config. 
Define it like this. Also define a converter class. 
```
    @Key("CLICKHOUSE_COMPRESSION_ALGORITHM")
    @DefaultValue("lz4")
    @ConverterClass(ClickHouseCompressionConvertor.class)
    ClickHouseCompression getClickhouseCompressionAlgorithm();
```